extension/lib/main.js

```diff
@@ -113,14 +110,65 @@ exports.main = function (options, callbacks) {
         widget.tooltip = config.extension.widget_tooltips[data];
         break;
     }
+  });*/
+
+  var { Frame } = require("sdk/ui/frame");
+  var { ToggleButton } = require('sdk/ui/button/toggle');
+  var { Toolbar } = require("sdk/ui/toolbar");
+
+  var frame = new Frame({
+    url: "./frame/frame.html"
   });
 
+  var menuButton = ToggleButton({
+    id: "menu-button",
+    label: "Menu",
+    icon: {
+      "32" : "./addon32.png",
+      "64" : "./addon64.png"
+    },
+    onChange: handleMenuButtonChange
+  });
+
+  var logButton = ToggleButton({
+    id: "log-button",
+    label: "Logging",
+    icon: {
+      "32" : "./addon32.png",
+      "64" : "./addon64.png"
+    },
+    onChange: handleLogButtonChange
+  });
+
+  var toolbar = Toolbar({
+    name: "memchaser-toolbar",
+    title: "MemChaser",
+    items: [frame, menuButton, logButton]
+  });
+
+  function handleMenuButtonChange(state) {
+    if (state.checked) {
+      contextPanel.show({
+        position: menuButton
+      });
+    }
+  }
+
+  function handleContextPanelHide() {
+    menuButton.state('window', {checked: false});
+  }
+
+  function handleLogButtonChange(state) {
+    logger.active = state.checked;
+    logButton.label = "Logging is " + (logger.active ? "enabled" : "disabled");
```

Comment:
We do not correctly initialize this value, so hovering over the button right after startup will only show 'logging'.